charts/promitor/templates/deployment.yaml

```diff
@@ -0,0 +1,75 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "promitor.fullname" . }}
+  labels:
+    app: {{ template "promitor.name" . }}
+    chart: {{ template "promitor.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 1
+  selector:
+    matchLabels:
+      app: {{ template "promitor.name" . }}
+      release: {{ .Release.Name }}
+      type: {{ .Values.service.selectorType }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "promitor.name" . }}
+        release: {{ .Release.Name }}
+        type: {{ .Values.service.selectorType }}
+    spec:
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - containerPort: {{ .Values.service.targetPort }}
+          env:
+          - name: PROMITOR_AUTH_APPID
+            valueFrom:
+              secretKeyRef:
+                  name: {{ template "promitor.fullname" . }}
+                  key: azure-app-id
+          - name: PROMITOR_AUTH_APPKEY
+            valueFrom:
+              secretKeyRef:
+                  name: {{ template "promitor.fullname" . }}
+                  key: azure-app-key
+          - name: PROMITOR_HTTP_PORT
+            value: {{ .Values.service.targetPort | quote }}
+          - name: PROMITOR_SCRAPE_BASEPATH
+            value: {{ .Values.scrapeConfig.path | quote }}
+          - name: PROMITOR_SCRAPE_SCHEDULE
+            value: {{ .Values.scrapeConfig.schedule | quote }}
+          - name: PROMITOR_FEATURE_METRICSTIMESTAMP
+            value: {{ .Values.scrapeConfig.timestamp | quote }}
+        {{- if .Values.telemetry.appInsightsKey }}
+          - name: PROMITOR_TELEMETRY_INSTRUMENTATIONKEY
+            valueFrom:
+              secretKeyRef:
+                  name: {{ template "promitor.fullname" . }}
+                  key: azure-appinsights-key
+        {{- end }}
+          volumeMounts:
+          - name: config-volume-{{ template "promitor.fullname" . }}
+            mountPath: /config/
+          livenessProbe:
+            httpGet:
+              path: /api/v1/health
+              port: {{ .Values.service.targetPort }}
+          readinessProbe:
+            httpGet:
+              path: /api/v1/health
+              port: {{ .Values.service.targetPort }}
```

Comment:
We could also use named ports - https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#use-a-named-port

Reply:
Since the liveness probe and prometheus will both be checking the same point, I'm not sure naming the ports will make things clearer. I can rename the variable in the values file to be "containerPort" "internalPort" or similar but making it a named port will just add in another variables that points to something that is already a variable.

Reply:
👌 it's ok as is, agree with your remarks